html/pfappserver/lib/pfappserver/Form/Config/Switch.pm

```diff
@@ -490,9 +490,41 @@ sub addRoleMapping {
 addRoleMapping("VlanMapping", "vlan");
 addRoleMapping("UrlMapping", "url");
 addRoleMapping("ControllerRoleMapping", "controller_role");
-addRoleMapping("AccessListMapping", "accesslist", [validate_method => \&_validate_acl ]);
+addRoleMapping("AccessListMapping", "accesslist", [validate_method => \&_validate_acl_switch ]);
 addRoleMapping("VpnMapping", "vpn");
 
+sub _validate_acl_switch {
+    my ($field) = @_;
+    my $switch = $field->form->getSwitch();
+    if ($switch) {
+       my $role_field = $field->parent()->field('role');
+       my $error = $switch->checkRoleACLs(
+           $role_field->value,
+           [split /\n/, $field->value],
+       );
+       if ($error) {
+           $field->add_error($error->{message});
+       }
+    }
+}
+
+sub getSwitch {
+    my ($self) = @_;
+    my $type_field = $self->field('type');
+    my $type = $type_field->value;
+    if (!defined $type) {
+        return undef;
+    }
+
+    my $value = $self->value;
+    my $module = pf::SwitchFactory::getModule($type);
+    if ($module->require() ) {
+        return $module->new($value);
```

Comment:
This doesn't return the switch config , all the field are empty and it fail after in the code.